source/features/pull-request-timeline-filter.tsx

```diff
@@ -0,0 +1,260 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+async function sleep(seconds: number): Promise<any> {
+	await new Promise(resolve => setTimeout(resolve, seconds * 1000));
+}
+
+interface FilterSettings {
+	HideUnresolved: boolean;
+	HideResolved: boolean;
+	hideNormalComment: boolean;
+	HideCommits: boolean;
+	AutoLoadHidden: boolean;
+	HideOthers: boolean;
+}
+
+const CurrentSettings: FilterSettings =
+{
+	HideResolved: true,
+	HideUnresolved: false,
+	hideNormalComment: false,
+	HideOthers: false,
+	HideCommits: false,
+	AutoLoadHidden: false
+};
```

Comment:
You should be able to read the values of each checkbox by referring to their `form` element and using FormData, I think, to read the values. For example, instead of having this global, have a function that returns `new FormData(select(the form))` and then access its values with `.get(id)`

Reply:
Alternatively they can also be read with `form.fields[name].checked`

Reply:
Done